### DISCUSSION TITLE
[05/12/2024] Score Community Meeting

### DISCUSSION THREAD

**Discussion by sujaya-sys** [score-spec/spec#109] [Community meetings]
Last community call of the year! 🎉

**Info**

| Community call | Info |
|:-----------|:------------|
| Meeting Link | Join via [Google Meet](https://meet.google.com/znt-usdc-hzs) or call +49 40 8081618260 (Pin: 599 887 196)
| Meeting Agenda & Notes | Add to our agenda or review minutes [here](https://github.com/score-spec/spec/discussions/categories/community-meetings)
| Meeting Time | 1:00-2:00pm UTC, every first Thursday of the month

**Open To Dos**

- [x] @sujaya-sys to create PR for [blog post](https://github.com/score-spec/score.dev/issues/16)
- [ ] @sujaya-sys to write up and discuss with Florian how to quantify or drive Score adoption (GHA action usage, CLIs usage, Website and Docs traffic)
- [x] @sujaya-sys to align with Ben and Andrej on next steps for Azure implementation 
- [ ] @sujaya-sys @chrishumanitec Implement decision making process on new issues as discussed [here](https://github.com/score-spec/spec/discussions/107#discussioncomment-11178592)
- [ ] @chrishumanitec  Create issue to turn files into key value pair rather than array @chrishumanitec

**Rollovers from the last call**

- Let's file issue(s) to support a new `score-* resources available-provisioners` command generating in the output this table or a more advanced/detailed version with `-o json`: https://docs.score.dev/docs/score-implementation/score-compose/resources-provisioners/? In `score-go` with that it's common across all the implementations?
- Let's add a new "Examples" entry in the docs where we port all of the examples from there https://github.com/score-spec/score-compose/tree/main/examples?
  - 1 page per Score file?
  - Where to store the Score files (because common to `score-compose` and `score-k8s`)? In a new dedicated `examples` repo?
  - Where do we show the `score-compose init|generate`+`compose up` or `score-k8s init|generate`+`kubectl apply -f`? Under this section https://docs.score.dev/docs/score-implementation/? Or on each example page with 2 tabs (`score-compose` and `score-k8s`)?

**Comment by sujaya-sys** [comment]
Hi all, unfortunately I'll be unable to join the call this time. Here are my updates:

#### Write up and discuss with Florian how to quantify or drive Score adoption (GHA action usage, CLIs usage, Website and Docs traffic)
Status: Outstanding, will reach out to Florian asap and have something by our next meeting latest

#### Implement decision making process on new issues as discussed https://github.com/score-spec/spec/discussions/107#discussioncomment-11178592
Status: Outstanding, I'll kick this off with a first topic asap.

#### Blog post
Status: Done! [Settling in - Score’s journey in the CNCF ecosystem](https://score.dev/blog/score-journey-in-the-cncf-ecosystem/)

#### Azure implementation
Status: Here's Andrej's response, let's align on a focus as a next step:
> From this I conclude that the Azure implementation should primarily focus on the “Azure Container Apps”. What is not yet clear to me is which target format should be implemented here. Azure offers various options for generating resources.
> - Azure Resource Manager (ARM)
> - Bicep
> - Azure Developer CLI (azd)
> - Azure CLI (manual deployments)
>
> Have you already talked about it? What is the preference?
>

**Reply by giesan** [reply]
Bicep is recommended by Microsoft itself.
See https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/syntax

**Reply by mathieu-benoit** [reply]
Discussed the Azure Container App versus Azure Container Instance: Azure Container Apps is the selected approach (free tier, etc.).

Thoughts:
1. Generate generic Bicep: https://learn.microsoft.com/en-us/azure/templates/microsoft.app/containerapps?pivots=deployment-language-bicep
2. Generate the YAML schema and then run `az containerapp create -n my-containerapp --yaml "path/to/yaml/file.yml"`

We will go with 1., and then people will do:
```
az deployment group create \
  --name demoRGDeployment \
  --resource-group ExampleGroup \
  --template-file main.bicep \
  --parameters storageAccountType=Standard_GRS
```

**Reply by mathieu-benoit** [reply]
@chrishumanitec to contact @giesan for next steps.

**Reply by mathieu-benoit** [reply]
Sequence to tackle:
1. Score spec as-is in Azure Container App Bicep --> Start with that.
2. "Azure Container App"'s Score extension for the rest (example annotations for `score-compose`)... "how to inject more specific things".
3. Default provisioners

**Comment by mathieu-benoit** [comment]
- Discussion around `score-k8s`'s `Deployment` manifest:
  - https://github.com/score-spec/score-k8s/issues/35
  - https://github.com/score-spec/score-k8s/issues/25
  - Should we work next on how this could be extended/patched? "I want something else than `Deployment`". "Workload patching", which language: CEL, etc.
  - Note: the drawback of this approach is that it will be less intuitive, and complicated to use if "everything is patched".
  - Also, quid of how to pass the info of "I want `StatefulSet` from Score to `score-k8s`".
  - Notion of "extension"
  - Notion of patch with Kustomize
